pkg/bootstrap/option/convert.go

```diff
@@ -135,13 +137,14 @@ func tlsContextConvert(tls *networkingAPI.ClientTLSSettings, sniName string, met
 			},
 		}
 		tlsContext.CommonTlsContext.AlpnProtocols = util.ALPNInMeshH2
+		// For ISTIO_MUTUAL if custom SNI is not provided, use the default SNI name.
+		if len(tls.Sni) == 0 {
+			tlsContext.Sni = sniName
```

Comment:
why do the other types use `tls.Sni` but here we use `sniName`?

Reply:
Good catch we have to set `tls.Sni` if it exists and if it does not exist for ISTIO_MUTUAL we will have to set the default SNI name. This was existing implementation

Reply:
Looks like for ISTIO_MUTUAL we only honor the sniName not the ones given in tls.Sni. So the existing implementation is correct.

Reply:
Ignore the above. I corrected the code. If `tls.Sni `  exists we use it otherwise use the default `sniName`